p2p/connectionpool/connectionpool.go

```diff
@@ -224,7 +232,10 @@ func (cp *ConnectionPool) GetConnection(ctx context.Context, address inet.Addr,
 	cp.pending[remotePub] = append(cp.pending[remotePub], pendChan)
 	if !found {
 		// No one is waiting for a connection with the remote peer, need to call Dial
-		cp.dialWait.Add(1)
+		cp.dialWaitMu.Lock()
+		cp.dialWait.Add(1) //  The first increment must be synchronized with Wait.
```

Comment:
what if Wait completes and we call Add(1) afterwards? i think this is the bug and we need to check if connection pool is closed before Add(1)